src/main/java/org/cardanofoundation/cip8/CIP8Parser.java

```diff
@@ -28,25 +28,65 @@
 import static org.cardanofoundation.ext.cbor.MoreCbor.serialize;
 import static org.cardanofoundation.ext.cose.COSEKey.deserialize;
 
+/**
+ * The {@code CIP8Parser} class is used to verify and parse a message that uses the CIP 8
+ * specification of the Cardano protocol.
+ * For further information on CIP 8, see <A href="https://cips.cardano.org/cips/cip8/">
+ * https://cips.cardano.org/cips/cip8/</A>.
+ */
 public final class CIP8Parser {
 
+    /**
+     * Static instance of {@code Logger} used for logging.
+     */
     private static final Logger logger = LoggerFactory.getLogger(CIP8Parser.class);
 
+    /**
+     * TODO
+     */
     private static final EdDSAParameterSpec ED_DSA_PARAMETER_SPEC = EdDSANamedCurveTable.getByName(ED_25519);
+
+    /**
+     * TODO
+     */
     private static final long PUBLIC_KEY_INDEX = -2;
 
+    /**
+     * Signature of the message signed using CIP 8.
+     */
     private final String signature;
+
+    /**
+     * Public key used to sign the message using CIP 8.
```

Comment:
Here I think it is worth to add that actual key is hex encoded and also mention why public key is optional, basically in typical scenarios I think public key is part of CIP-8 signature but there are cases where public key is not encoded in signature itself and then a developer / client can pass public key explicitly.